silhouette/app/com/mohiva/play/silhouette/impl/providers/state/CsrfStateItemHandler.scala

```diff
@@ -35,7 +35,7 @@ class CsrfStateItemHandler @Inject() (
    */
   override def item(implicit ec: ExecutionContext): Future[Item] = {
     idGenerator.generate.map { id =>
-      CsrfState(id)
+      CsrfState(cookieSigner.sign(id))
```

Comment:
That makes, hence I was confused about signing it multiple times. Thanks for catching it.